app/models/mesh/Mesh.scala

```diff
@@ -0,0 +1,143 @@
+package models.mesh
+
+import com.google.common.io.BaseEncoding
+import com.scalableminds.util.accesscontext.DBAccessContext
+import com.scalableminds.util.geometry.Point3D
+import com.scalableminds.util.tools.Fox
+import com.scalableminds.webknossos.schema.Tables._
+import controllers.TaskParameters
+import javax.inject.Inject
+import play.api.libs.json._
+import slick.jdbc.PostgresProfile.api._
+import slick.lifted.Rep
+import utils.{ObjectId, SQLClient, SQLDAO}
+import play.api.libs.functional.syntax._
+import play.api.libs.json.Json._
+import play.api.libs.json._
+
+import scala.concurrent.ExecutionContext
+
+case class MeshInfo(
+  _id: ObjectId,
+  _annotation: ObjectId,
+  description: String,
+  position: Point3D,
+  created: Long = System.currentTimeMillis,
+  isDeleted: Boolean = false
+)
+
+case class MeshInfoParameters(
+                               annotationId: ObjectId,
+                               description: String,
+                               position: Point3D,
+                             )
+object MeshInfoParameters {
+  implicit val meshInfoParametersReads: Reads[MeshInfoParameters] =
+    ((__ \ "annotationId").read[String] (ObjectId.stringObjectIdReads("teamId")) and
+      (__ \ "description").read[String] and
+      (__ \ "position").read[Point3D])((annotationId, description, position) => MeshInfoParameters(ObjectId(annotationId), description, position))
+}
+
+
+class MeshService @Inject()(meshDAO: MeshDAO)(implicit ec: ExecutionContext) {
+  def publicWrites(meshInfo: MeshInfo): Fox[JsObject] = {
+    Fox.successful(Json.obj(
+      "_id" -> meshInfo._id.toString,
+      "annotation" -> meshInfo._annotation.toString,
+      "description" -> meshInfo.description,
+      "position" -> meshInfo.position
+    ))
+  }
+
+  def compactWrites(meshInfo: MeshInfo): Fox[JsObject] = {
+    Fox.successful(Json.obj(
+      "_id" -> meshInfo._id.toString,
+      "description" -> meshInfo.description,
+      "position" -> meshInfo.position
+    ))
+  }
+}
+
+class MeshDAO @Inject()(sqlClient: SQLClient)(implicit ec: ExecutionContext) extends SQLDAO[MeshInfo, MeshesRow, Meshes](sqlClient) {
+  val collection = Meshes
+
+  def idColumn(x: Meshes): Rep[String] = x._Id
+
+  def isDeletedColumn(x: Meshes): Rep[Boolean] = x.isdeleted
+
+  def infoColumns = (columnsList diff Seq("data")).mkString(", ")
+  type InfoTuple = (String, String, String, String, java.sql.Timestamp, Boolean)
+
+  override def readAccessQ(requestingUserId: ObjectId): String = s"""
+    (isPublic or _team in (select _team from webknossos.user_team_roles where _user = '${requestingUserId.id}') or _user = '${requestingUserId.id}'
```

Comment:
Could it be that the identifiers `isPublic` and `_team` are not correctly used here? If trying to add mesh meta data, I get the following error:

```
<~ SQL Failure: ERROR: column "isPublic" does not exist Position: 119
```

Reply:
Actually, I think I was meaning to remove it entirely anyway, since the access checks go via the Annotation instead now.